performance/autoupdate-clients/main.go

```diff
@@ -0,0 +1,258 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+	"os/signal"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+)
+
+func main() {
+	N := getNFromCommandLine()
+	host, user, pw := getParams()
+	jar := getCookieJar(user, pw, host)
+
+	sumBytes := 0
+	oldSumKB := 0
+	changeIds := make(map[int]int)
+	changeIdsMutex := &sync.Mutex{}
+	gotUpdate := false
+
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
```

Comment:
I saw this TODO
>Pressing Ctrl+C during connecting should stop the process immediatly

If you move this line after the connection, you would solve this TODO